subsys/bluetooth/host/conn.c

```diff
@@ -1205,28 +1207,37 @@ void bt_conn_set_state(struct bt_conn *conn, bt_conn_state_t state)
 	}
 }
 
-struct bt_conn *bt_conn_lookup_handle(uint16_t handle)
+struct bt_conn *bt_conn_lookup_handle(uint16_t handle, int type)
 {
 	struct bt_conn *conn;
 
 #if defined(CONFIG_BT_CONN)
 	conn = conn_lookup_handle(acl_conns, ARRAY_SIZE(acl_conns), handle);
 	if (conn) {
-		return conn;
+		if (type & conn->type) {
+			return conn;
+		}
+		LOG_WRN("incompatible handle %u", handle);
 	}
 #endif /* CONFIG_BT_CONN */
 
 #if defined(CONFIG_BT_ISO)
 	conn = conn_lookup_handle(iso_conns, ARRAY_SIZE(iso_conns), handle);
 	if (conn) {
-		return conn;
+		if (type & conn->type) {
+			return conn;
+		}
+		LOG_WRN("incompatible handle %u", handle);
```

Comment:
This part should at most be an assert, I think, since `iso_conns` should only contain BT_CONN_TYPE_ISO objects.

Reply:
We search through the list regardless of the passed type, though.  We could make it more like `bt_conn_foreach` iterating only over the correct lists (or even use the foreach function, but all my ideas involve `setjmp`, which is not quite pretty either), but then it would miss the warnings and just return the NULL.